### PR TITLE
Finalize UG v1.2

### LINKED ISSUE #49: [UG] Finalize UG v1.2
* tidy up the commands
* Looks for any UG bugs
* Finish the command summary

### LINKED ISSUE #73: [BUG] Table of content in UG and DG not working
* Expected : table of content is rendered 
* image
* Result : 
* image

### PR DESCRIPTION
**Pull request by simonjulianl**
### Summary

* Finalize User Guide, some changes in format to make them uniform, Resolves #49 
* Resolve the bug that TOC not working, Resolves #73 

### Detail

* some parameter format change, including how to add patient 
* some minor comment changes in the tagging 
* change all optional parameter using [OPTIONAL_PARAMETER]
* change all flags into parameters (naming) 

### Checks

- [x] Link PR to issue
- [x] Add Reviewer
- [x] Pass CI

### CODE REVIEW

**Comment by ngsengleng** [comment]
The section on suggestions and help looks fine, ill get round to fleshing it out with mockups later. Waiting on the rest to check their parts.

**Comment by simonjulianl** [comment]
> The section on suggestions and help looks fine, ill get round to fleshing it out with mockups later. Waiting on the rest to check their parts.

Yup sure, by the way @ngsengleng I have changed the image for help to reflect our site, unless you wanna change the design for help window

File: docs/UserGuide.md [hunk 1]
```diff
@@ -266,44 +309,47 @@ Examples:
...
 * `a/AGE`: Matches the age field (Valid for: `Patients`)
 * `g/GENDER`: Matches the gender field; The only valid keywords for this field are `M/F/O` (Valid for: `Patients`)
 * `h/HEIGHT`: Matches the height field (Valid for: `Patients`)
 * `w/WEIGHT`: Matches the weight field (Valid for: `Patients`)
-* `b/BLOOD_TYPE`: Matches the blood type field; The only valid keywords for this field are `A/B/AB/O` 
-(Valid for: `Patients`)
-* `m/MEDICAL_CONDITION`: Limits the keyword search to the list of medical conditions of a patient (Valid for: `Patients`)
+* `b/BLOOD_TYPE`: Matches the blood type field; The only valid keywords for this field are `A/B/AB/O`
+  (Valid for: `Patients`)
+* `m/MEDICAL_CONDITION`: Limits the keyword search to the list of medical conditions of a patient (Valid
```
Comment (yyhangz) [inline docs/UserGuide.md:334]: Should we change this parameter flag to `o/` to be consistent with the `add patient` command?

```suggestion
* `o/MEDICAL_CONDITION`: Limits the keyword search to the list of medical conditions of a patient (Valid
```

**Review by yyhangz — COMMENTED** [review]
Made a suggestion for the `find` command parameter flag for patient medical conditions. Otherwise, the changes on my parts about `tag` and `find` are fine imo. Thanks Simon!!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -266,44 +309,47 @@ Examples:
...
+* Parameters can be repeated (e.g. `find n/Hans n/Bo` will return both `Hans Gruber` and `Bo Yang`)
+* If the optional parameters is not specified, the keyword will match any fields. E.g. `find dia` will return:
     1. Doctor Claudia, whose name matches `dia`
     2. Patient Jaryl, whose medical condition, `diabetes`, matches `dia`
     3. Doctor Tom, whose specialty, `Pediatrics`, matches `dia`
     4. Patient Lydia, whose name matches `dia`
 
 Examples:
+
 * `find m/diabetes a/42 n/Jaryl`
```
Comment (yyhangz) [inline docs/UserGuide.md:353]: Likewise, if the previous suggestion for the parameter flag of patients' medical conditions is changed, then we can update this example as well.

```suggestion
* `find o/diabetes a/42 n/Jaryl`
```

File: docs/UserGuide.md [hunk 3]
```diff
@@ -144,52 +158,58 @@ Examples:
...
 * Invalid Patient ID being supplied would be flagged by GoMedic, and do not cause changes to any existing patients.
-* Invalid `OPTIONAL_TO_DELETE` conditions supplied would be flagged by GoMedic, and do not cause changes to the
-existing patient.
+* Invalid `OPTIONAL_TO_DELETE` conditions supplied would be flagged by GoMedic, and do not cause changes to the existing
+  patient.
 
 Examples:
 
 * `update t/patient i/P123 n/John-Doe a/30 g/M`
 * `update t/patient i/P003 n/Tom-Doe a/20 g/M h/167 w/61 b/AB c/12341234 do/diabetes`
```
Comment (radiankrisno) [inline docs/UserGuide.md:191]: Can add example that deletes 2 medical conditions and add medical condition as well to give a better idea to the user.
e.g. update t/patient i/P003 n/Tom-Doe a/20 g/M h/167 w/61 b/AB c/12341234 do/diabetes do/heart-feailure

**Review by radiankrisno — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 1 above)
Comment (simonjulianl) [reply docs/UserGuide.md]: this sounds good

File: docs/UserGuide.md (hunk 2 above)
Comment (simonjulianl) [reply docs/UserGuide.md:353]: yup sure

File: docs/UserGuide.md (hunk 3 above)
Comment (simonjulianl) [reply docs/UserGuide.md:195]: yup sure, I add also combination with o/ flag

**Review by radiankrisno — APPROVED** [review]
LGTM

**Pull request merged**